src/main/java/homework_nr_18/TaskManager.java

```diff
@@ -0,0 +1,58 @@
+package homework_nr_18;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TaskManager {
+    public String task;
+    protected List<TaskManager> taskList = new ArrayList<>();
+    
+    public TaskManager(String task) {
+        if (task != "" & task != null) this.task = task;
+    }
+    
+    public TaskManager() {
+    
+    }
+    
+    public void addTask(String task) {
+        synchronized (task) {
+            if (task != "" & task != null)
+                taskList.add(new TaskManager(task));
+        }
+    }
+    
+    public void removeTask(String task) {   //!!!?????????
+        try {
+            Thread.sleep(50);
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
Rethrow the exception.
`throw new RuntimeException(e)`

If you get a InterruptedException then the thread was already interrupted so no need to interrupt again